tst/com/amazon/corretto/crypto/provider/test/FipsStatusTest.java

```diff
@@ -0,0 +1,41 @@
+package com.amazon.corretto.crypto.provider.test;
+
+import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
+import com.amazon.corretto.crypto.provider.FipsStatusException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+import org.junit.jupiter.api.parallel.ResourceAccessMode;
+import org.junit.jupiter.api.parallel.ResourceLock;
+
+import javax.crypto.KeyGenerator;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@ExtendWith(TestResultLogger.class)
+@Execution(ExecutionMode.CONCURRENT)
+@ResourceLock(value = TestUtil.RESOURCE_GLOBAL, mode = ResourceAccessMode.READ)
+public class FipsStatusTest {
+
+    private final AmazonCorrettoCryptoProvider provider = AmazonCorrettoCryptoProvider.INSTANCE;
+
+    @Test
+    public void givenAccpBuiltWithFips_whenAWS_LC_fips_failure_callback_expectException() throws Exception {
+        if (provider.isFips()) {
```

Comment:
what's the expected behavior when `isFipsStatusOk()` is called on a non-FIPS build?


is there anything we should test in the non-FIPS case, if only to codify that expected behavior?

Reply:
For non-FIPs builds, it always returns true since AWS-LC doesn't do callback registration. We also don't check for instantiate service classes